server/routes/api/protein.py

```diff
@@ -13,22 +13,411 @@
+            # add both interaction dcids and their reversals
+            interaction_dcid_set.update(interaction_dcids_filtered)
+            interaction_dcid_set.update(
+                map(_reverse_interaction_dcid, interaction_dcids_filtered))
+
+            source_id = _id(source_dcid)
+            target_ids = []
+            for interaction_dcid in interaction_dcids_filtered:
+                try:
+                    target_id = _other_interactor(interaction_dcid, source_id)
+                except ValueError:
+                    continue
+                target_ids.append(target_id)
+            # getter for score of source-target interaction
+            target_scorer = lambda target_id: scores.get(
+                f'{source_id}_{target_id}', DEFAULT_INTERACTION_SCORE)
+            # true if score-target interaction has score above threshold
+            target_ids_filtered = filter(
+                lambda target_id: target_scorer(target_id) > score_threshold,
+                target_ids)
+            # sort in descending order of score
+            target_ids_sorted = sorted(target_ids_filtered,
+                                       key=target_scorer,
+                                       reverse=True)
+            partition = _partition_expansion_cross(target_ids_sorted,
+                                                   node_id_set)
+            expansion_target_ids = partition['expansion_targets']
+            cross_target_ids = partition['cross_targets']
+            expansion_target_ids = expansion_target_ids[:max_interactors]
+            target_link_getter = lambda target_id: {
+                'source': source_id,
+                'target': target_id,
+                'score': target_scorer(target_id)
+            }
+            # add cross links to links in last layer
+            # (E.g. depth 1 graph will show center, neighbors of center, and cross-links between neighbors)
+            links[-1].extend(map(target_link_getter, cross_target_ids))
+            # TODO: track distance of each node to center
+            new_node_ids.update(expansion_target_ids)
+
+            # final iteration expansion links are extra, shouldn't get rendered
+            if depth != max_depth + 1:
+                expansion_links.extend(
+                    map(target_link_getter, expansion_target_ids))
+                last_layer_node_dcids.extend(map(_dcid, expansion_target_ids))
+
+        if depth != max_depth + 1:
+            new_nodes = []
+            for new_node_id in new_node_ids:
+                try:
+                    new_node = _node(new_node_id, depth)
+                except ValueError:
+                    continue
+                new_nodes.append(new_node)
+            nodes.append(new_nodes)
+            node_id_set.update(new_node_ids)
+            links.append(expansion_links)
+
+    graph = {'nodeDataNested': nodes, 'linkDataNested': links}
+
+    return Response(json.dumps(graph))
```

Comment:
Also need to return a 200 code: https://github.com/datacommonsorg/website/blob/master/server/routes/api/stats.py#L539

Reply:
Thanks will do!